sqlserver/datadog_checks/sqlserver/metrics.py

```diff
@@ -148,92 +141,104 @@ def fetch_all_values(cls, cursor, counters_list, logger, databases=None):
-                base = cval2
-            else:
-                value = cval2
-                base = cval
+            key = '{}::{}'.format(numerator['instance_name'], numerator['object_name'])
+            corresponding_base = base_by_key.get(key)
+
+            if not corresponding_base:
+                self.log.warning(
+                    'Could not find corresponding base counter for sql_name: %s base_name: %s',
+                    self.sql_name,
+                    self.base_name,
+                )
 
             metric_tags = list(self.tags)
             if self.instance == ALL_INSTANCES:
-                metric_tags.append('{}:{}'.format(self.tag_by, inst.strip()))
-            self.report_fraction(value, base, metric_tags)
+                metric_tags.append('{}:{}'.format(self.tag_by, instance_name))
+            self.report_fraction(
+                numerator['cntr_value'], corresponding_base['cntr_value'], metric_tags, previous_values=values_cache
+            )
 
-    def report_fraction(self, value, base, metric_tags):
+    def report_fraction(self, value, base, metric_tags, previous_values):
         try:
             result = value / float(base)
-            self.report_function(self.datadog_name, result, tags=metric_tags)
+            self.report_function(self.metric_name, result, tags=metric_tags)
         except ZeroDivisionError:
-            self.log.debug("Base value is 0, won't report metric %s for tags %s", self.datadog_name, metric_tags)
+            self.log.debug("Base value is 0, won't report metric %s for tags %s", self.metric_name, metric_tags)
 
 
 class SqlIncrFractionMetric(SqlFractionMetric):
-    def report_fraction(self, value, base, metric_tags):
-        key = "key:" + "".join(metric_tags)
-        if key in self.past_values:
-            old_value, old_base = self.past_values[key]
+    """
+    Performance counters where the cntr_type column value is 1073874176 display
+    how many items are processed on average, as a ratio of the items processed
+    to the number of operations. For example, the Locks:Average Wait Time (ms)
+    counters compares the lock waits per second with the lock requests per second,
+    to display the average amount of wait time (in milliseconds) for each lock request that resulted in a wait.
+    As such, to get a snapshot-like reading of the last second only, you must compare the delta between
+    the current value and the base value (denominator) between two collection points that are one second apart.
+    """
+
+    def report_fraction(self, value, base, metric_tags, previous_values):
+        # return if nil is passed as the values cache, as this should be instantiated
+        # at check instantiation
+        if previous_values is None:
+            return
+        # key is set to the metric name + the metric tags in order to support
+        # per database instance fraction metrics
+        key = "{}:{}".format(self.metric_name, "".join(metric_tags))
```

Comment:
prior to this change, the key was set to only the metric tags which meant if we reported multiple sqlserver inc metrics the values would overwrite each other and things would get reported incorrectly